pyterrier_dr/bge_m3.py

```diff
@@ -0,0 +1,77 @@
+from pyterrier.transformer import Transformer
+from FlagEmbedding import BGEM3FlagModel
+from tqdm import tqdm
+import pyterrier as pt
+import pandas as pd
+import numpy as np
+import torch
+from .biencoder import BiEncoder
+
+
+class BGEM3Factory(BiEncoder):
```

Comment:
The convention in the -dr package is to omit the "Factory" suffix -- it's a bit cleaner.